SayTheirNamesTests/ServiceTests/DeepLinkTests.swift

```diff
@@ -0,0 +1,161 @@
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+import XCTest
+@testable import Say_Their_Names
+
+/// Unit Tests for the Deep Linking mechanism.
+/// The test cases rely on properly building a set of `DeepLink`s
+/// from the DeepLinkHandler's `handle(urls:)` method.
+///
+class DeepLinkTests: XCTestCase {
+
+    // The URLs that will be used to construct the deep links.
+    var dlAbout: URL!
+    var dlPeopleJaoP: URL!
+    var dlPetitions: URL!
+    var dlDonations: URL!
+    var dlPetitionScurlock: URL!
+    var dlDonationFloydG: URL!
+    
+    // The scheme and host that should be used.
+    let dlScheme = "stn"
+    let dlHost = "saytheirname.netlify.app"
+    
+    // The relative paths for deep link URLs
+    let dlPathAbout = "about"
+    let dlPathPeopleJaoP = "profile/joao-pedro"
+    let dlPathPetitions = "petitions"
+    let dlPathDonations = "donations"
+    let dlPathPetitionScurlock = "petitions/petition-for-james-scurlock"
+    let dlPathDonationFloydG = "donate/donate-to-george-floyd"
+    
+    var container: DependencyContainer!
+
+    override func setUpWithError() throws {
+
+        // We setup all the deep link URLs. An exception is thrown is the URL
+        // can't be built from the given string.
+        dlAbout = try XCTUnwrap(URL(string: dlScheme + "://" + dlHost + "/" + dlPathAbout))
+        dlPeopleJaoP = try XCTUnwrap(URL(string: dlScheme + "://" + dlHost + "/" + dlPathPeopleJaoP))
+        dlPetitions = try XCTUnwrap(URL(string: dlScheme + "://" + dlHost + "/" + dlPathPetitions))
+        dlDonations = try XCTUnwrap(URL(string: dlScheme + "://" + dlHost + "/" + dlPathDonations))
+        dlPetitionScurlock = try XCTUnwrap(URL(string: dlScheme + "://" + dlHost + "/" + dlPathPetitionScurlock))
+        dlDonationFloydG = try XCTUnwrap(URL(string: dlScheme + "://" + dlHost + "/" + dlPathDonationFloydG))
+
+        // We instantiate the dependency container that will be used to build
+        // the various `DeepLink` objects, via `.deepLinkHandler.handle(urls:)`
+        container = DependencyContainer()
```

Comment:
You could just initialize `DeepLinkHandler`.

Reply:
Same as above. Not public. But I'll change the permissions and add the file to the test target.

Reply:
This however was a quickly unraveling... Since `DependencyContainer()` is final, I don't need to import anything to use it. However when I start using `DeepLinkHandler` directly, I have to pull **a lot** of files into the test target. So I decided against it considering the cost vs. reward.